net-misc/aria2/aria2-1.35.0.ebuild

```diff
@@ -0,0 +1,184 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI="7"
+
+inherit bash-completion-r1
+
+DESCRIPTION="A download utility with segmented downloading with BitTorrent support"
+HOMEPAGE="https://aria2.github.io/"
+SRC_URI="https://github.com/aria2/${PN}/releases/download/release-${PV}/${P}.tar.xz"
+
+LICENSE="GPL-2+-with-openssl-exception"
+KEYWORDS="~amd64 ~arm ~arm64 ~ppc ~ppc64 ~sparc ~x86 ~amd64-linux ~x86-linux"
+SLOT="0"
+IUSE="adns bittorrent +gnutls jemalloc libuv +libxml2 metalink +nettle nls sqlite scripts ssh ssl tcmalloc test xmlrpc"
+# xmlrpc has no explicit switch, it's turned out by any XML library
+# so metalink implicitly forces it on
+REQUIRED_USE="?? ( jemalloc tcmalloc )
+	metalink? ( xmlrpc )"
+RESTRICT="!test? ( test )"
+# Crazy GnuTLS/OpenSSL/etc. logic below:
+# 1. Those libraries are used for two purposes: TLS & MD.
+# 2. Upstream preferences are:
+#    1) gnutls [tls]
+#    2) !gnutls? -> openssl [tls+md]
+#    3) !openssl? -> nettle [md]
+#    4) !openssl? !nettle? -> gcrypt [md]
+#    5) !*? -> bundled md routines (we don't use them)
+# 3. There's also gmp usage for bittorrent with nettle.
+# 4. You can't really control features, just dependencies.
+# (we are skipping native Apple/Windows TLS support)
+#
+# We map this into:
+# ssl? -> openssl || (gnutls + (nettle || libgcrypt ))
+# !ssl? -> nettle || libgcrypt
+
+RDEPEND="sys-libs/zlib:0=
+	adns? ( >=net-dns/c-ares-1.5.0:0= )
+	jemalloc? ( dev-libs/jemalloc )
+	libuv? ( >=dev-libs/libuv-1.13:0= )
+	metalink? (
+		libxml2? ( >=dev-libs/libxml2-2.6.26:2= )
+		!libxml2? ( dev-libs/expat:0= ) )
+	sqlite? ( dev-db/sqlite:3= )
+	ssh? ( net-libs/libssh2:= )
+	ssl? (
+		app-misc/ca-certificates
+		gnutls? (
+			>=net-libs/gnutls-1.2.9:0=
+			nettle? (
+				>=dev-libs/nettle-2.4:0=
+				bittorrent? (
+					>=dev-libs/nettle-2.4:0=[gmp]
+					>=dev-libs/gmp-6:0=
+				)
+			)
+			!nettle? ( >=dev-libs/libgcrypt-1.2.2:0= )
+		)
+		!gnutls? (
+			dev-libs/openssl:0=
+		)
+	)
+	!ssl? (
+		nettle? (
+			>=dev-libs/nettle-2.4:0=
+			bittorrent? (
+				>=dev-libs/nettle-2.4:0=[gmp]
+				>=dev-libs/gmp-6:0=
+			)
+		)
+		!nettle? ( >=dev-libs/libgcrypt-1.2.2:0= )
+	)
+	tcmalloc? ( dev-util/google-perftools )
+	xmlrpc? (
+		libxml2? ( >=dev-libs/libxml2-2.6.26:2= )
+		!libxml2? ( dev-libs/expat:0= ) )"
+
+DEPEND="${RDEPEND}
+	test? ( >=dev-util/cppunit-1.12.0:0 )"
+RDEPEND+="
+	nls? ( virtual/libiconv virtual/libintl )
+	scripts? ( dev-lang/ruby )"
+BDEPEND="app-arch/xz-utils
+	virtual/pkgconfig
+	nls? ( sys-devel/gettext )"
+
+pkg_setup() {
+	if use scripts && ! use xmlrpc; then
+		ewarn "Please note that you may need to enable USE=xmlrpc to run the aria2rpc"
+		ewarn "and aria2mon scripts against the local aria2."
+	fi
+}
+
+src_prepare() {
+	default
+	sed -i -e "s|/tmp|${T}|" test/*.cc test/*.txt || die "sed failed"
+	# Fails with USE=bittorrent && FEATURES=network-sandbox
+	sed -i -E \
```

Comment:
I'm sorry for missing this previously but I don't think you're actually using extended regexp syntax here.

Reply:
You're correct. I was experimenting with a one-liner to start with, but felt better with a readable solution.